lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@riccardonar yeah so check it out -- I have a LOWER version (8.4.20) -- and client_encoding works fine for me.

![screen shot 2015-04-02 at 4 30 18 pm](https://cloud.githubusercontent.com/assets/620091/6973047/f3b583d6-d955-11e4-8893-5baee2adea7d.png)

This just confirms what I was trying to tell you and @deeky666 -- this is not reliable data to make a decision off of.

@deeky666 confirmed he was just guessing anyway -- so we're leaving the current implementation (my fork). You can use my fork meanwhile until they release the next minor if you want.